theano/sandbox/cuda/tests/test_cula.py

```diff
@@ -26,44 +26,96 @@
 
 
 class TestCula(unittest.TestCase):
+    def setUp(self):
+        utt.seed_rng()
+
     def run_gpu_solve(self, A_val, x_val):
         b_val = numpy.dot(A_val, x_val)
         A = theano.tensor.matrix("A", dtype="float32")
-        b = theano.tensor.matrix("b", dtype="float32")
-
+        if x_val.ndim == 1:
+            b = theano.tensor.vector("b", dtype="float32")
+        else:
+            b = theano.tensor.matrix("b", dtype="float32")
```

Comment:
You can use this more generic code:
b = theano.tensor.TensorType(dtype="float32", broadcastable=[False,]*x_val.ndim)()